torch/nn/parameter.py

```diff
@@ -42,3 +46,82 @@ def __reduce_ex__(self, proto):
             torch._utils._rebuild_parameter,
             (self.data, self.requires_grad, OrderedDict())
         )
+
+
+class UninitializedParameter(Parameter):
```

Comment:
You want to add an entry to `docs/source/nn.rst` next to `Parameter` to make sure that this will be added to the doc properly.